diff -Naur abseil-cpp-20210324.2.org/absl/debugging/internal/examine_stack.cc abseil-cpp-20210324.2.sw/absl/debugging/internal/examine_stack.cc
--- abseil-cpp-20210324.2.org/absl/debugging/internal/examine_stack.cc	2022-03-10 01:13:54.492357080 +0000
+++ abseil-cpp-20210324.2.sw/absl/debugging/internal/examine_stack.cc	2022-03-10 01:15:26.202357080 +0000
@@ -46,7 +46,7 @@
     ucontext_t* context = reinterpret_cast<ucontext_t*>(vuc);
 #if defined(__aarch64__)
     return reinterpret_cast<void*>(context->uc_mcontext.pc);
-#elif defined(__alpha__)
+#elif defined(__alpha__) || defined(__sw_64__)
     return reinterpret_cast<void*>(context->uc_mcontext.sc_pc);
 #elif defined(__arm__)
     return reinterpret_cast<void*>(context->uc_mcontext.arm_pc);
