diff -uprN a/absl/base/options.h b/absl/base/options.h
--- a/absl/base/options.h	2025-02-03 23:50:45.000000000 +0800
+++ b/absl/base/options.h	2025-07-02 17:13:16.118136800 +0800
@@ -173,9 +173,11 @@
 // User code should not inspect this macro.  To check in the preprocessor if
 // absl::variant is a typedef of std::variant, use the feature macro
 // ABSL_USES_STD_VARIANT.
-
+#if defined(ARK_UI_X)
+#define ABSL_OPTION_USE_STD_VARIANT 0
+#else
 #define ABSL_OPTION_USE_STD_VARIANT 2
-
+#endif
 // ABSL_OPTION_USE_STD_ORDERING
 //
 // This option controls whether absl::{partial,weak,strong}_ordering are
