diff -uprN a/absl/debugging/symbolize.cc b/absl/debugging/symbolize.cc
--- a/absl/debugging/symbolize.cc	2025-02-03 23:50:45.000000000 +0800
+++ b/absl/debugging/symbolize.cc	2025-06-05 15:55:46.561863000 +0800
@@ -20,7 +20,7 @@
 #include <winapifamily.h>
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 // UWP doesn't have access to win32 APIs.
-#define ABSL_INTERNAL_HAVE_SYMBOLIZE_WIN32
+// #define ABSL_INTERNAL_HAVE_SYMBOLIZE_WIN32
 #endif
 #endif
 